api/v1alpha3/machine_types.go

```diff
@@ -89,6 +89,12 @@ type MachineSpec struct {
 	// Must match a key in the FailureDomains map stored on the cluster object.
 	// +optional
 	FailureDomain *string `json:"failureDomain,omitempty"`
+
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// +optional
+	NodeDrainTimeout int64 `json:"nodeDrainTimeout,omitempty"`
```

Comment:
Shouldn't this be a metav1.Duration?

Reply:
Yes, it can also be metav1.Duration. 
Do we have any significant advantage if now we change it from `int64` to `metav1.Duration` since it can lead to quite a lot of change to this PR?

Reply:
We typically use Duration for ... durations. The significant advantage is getting this right early on instead of having to make an API change later. @vincepri @CecileRobertMichon @detiber WDYT?

Reply:
Yes please, that's a good point. We should try to be consistent with the rest of the codebase

Reply:
I see. Fixed!